app/uk/gov/hmrc/testuser/services/Generator.scala

```diff
@@ -213,3 +214,38 @@ class PensionSchemeAdministratorIdentifierGenerator(random: Random = new Random)
     PensionSchemeAdministratorIdentifier(s"$initialCharacter$remainingDigits")
   }
 }
+
+object VrnChecksum {
+
+  def apply(s: String) = s + calcCheckSum97(weightedTotal(s))
+
+  def isValid(vrn: String): Boolean = {
+    if (regexCheck(vrn)) {
+      val total = weightedTotal(vrn)
+      val checkSumPart = takeCheckSumPart(vrn)
+      if (checkSumPart == calcCheckSum97(total).toInt) true
+      else checkSumPart == calcCheckSum9755(total)
+    } else false
+  }
+
+  private def calcCheckSum97(total: Int): String = {
+    var x = total
+    while (x >= 0) {
```

Comment:
I think this mutable x value and the while loop could be better expressed as a mathematical formula, something like:

val x = total % 97 - 97